src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -225,6 +225,14 @@ void BDXDownloader::PollTransferSession()
     } while (outEvent.EventType != TransferSession::OutputEventType::kNone);
 }
 
+void BDXDownloader::CleanupOnError(OTAChangeReasonEnum reason)
+{
+    Reset();
+    mBdxTransfer.Reset();
+    SetState(State::kIdle, reason);
+    mImageProcessor->Abort();
```

Comment:
I wonder if we can just move these three calls into `Reset`? I think most places where `Reset` is being called probably needs and/or already are making these same calls?